pandas/core/indexes/category.py

```diff
@@ -341,9 +341,26 @@ def __array__(self, dtype=None):
 
     @Appender(_index_shared_docs['astype'])
     def astype(self, dtype, copy=True):
+        if isinstance(dtype, compat.string_types) and dtype == 'category':
```

Comment:
I don't think you actually need this check as the ``dtype == self.dtype`` line below should pick this up.

Reply:
I think this, or some variation of it, is necessary in order to guarantee that `CI.astype('category')` doesn't change anything.  The issue being that `pandas_dtype('category')` returns `CDT(None, False)`, which would change a `CI` with `ordered=True` to `False`.